### PR TITLE
feat: added sas-js-date-days conversion

### PR DESCRIPTION
**Pull request by medjedovicm**
## Intent

Added conversion functions for SAS Date/Datetime to/from JS Date

A JS Date contains the number of _milliseconds_ since 01/01/1970
A SAS Date contains the number of _days_ since 01/01/1960
A SAS Datetime contains the number of _seconds_ since 01/01/1960

These functions calculate and compensate those differences and return correct result.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
- [x] All unit tests are passing (`npm test`).
- [ ] All `sasjs-cli` unit tests are passing (`npm test`).  **No existing function has beed touched**
- [x] Reviewer is assigned.

### CODE REVIEW

File: src/conversion/convertSasDaysToJsDate.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export const convertSasDaysToJsDate = (
```
Comment (allanbowe) [inline src/conversion/convertSasDaysToJsDate.ts]: needs a header like the other one

File: src/conversion/convertJsDateToSasDate.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * @param unit Unit in which to return the SAS Date / datetime, eg `sasdate | sasdatetime`
+ * @returns SAS Date value based on `unit` param
+ */
+export const convertJsDateToSasDate = (
+  jsDate: Date,
+  unit: 'sasdate' | 'sasdatetime'
+): number => {
+  let valueInMilliseconds: number = 0
+
+  valueInMilliseconds = new Date(
```
Comment (sabhas) [inline src/conversion/convertJsDateToSasDate.ts:18]: no need for  above declaration of variable at line # 16 just  add const here before variable

File: src/conversion/convertJsDateToSasDate.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  const msInTenYears = 315619200000 //calculating is not guaranteed to give the correct value, as this is parsed with: new Date(Date.UTC(1960, 0, 1)).getTime()
+
+  const ms1960 = valueInMilliseconds + msInTenYears
+
+  switch (unit) {
+    case 'sasdate': {
+      // always in days from 1960
+      let valueInDays = ms1960 / msInDay
+
+      valueInDays = Math.abs(valueInDays)
```
Comment (sabhas) [inline src/conversion/convertJsDateToSasDate.ts:39]: what's the reason for using absolute function? We can see value is already positive.

File: src/conversion/convertSasDateToJsDate.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    let jsMs = msNegativeTenYears + ms1960
+    let jsDate = new Date(jsMs)
+
+    let timezoneOffsetMs = jsDate.getTimezoneOffset() * 60 * 1000
+    jsMs += timezoneOffsetMs
+
+    jsDate = new Date(jsMs) // We create new Date object after the timezone correction
+
+    let jsDateUtcMs = Date.UTC(
```
Comment (sabhas) [inline src/conversion/convertSasDateToJsDate.ts:32]: change let to const.

Comment (sabhas) [reply src/conversion/convertSasDateToJsDate.ts:32]: try to use const at with all variable that are not being modified.

**Review by allanbowe — APPROVED** [review]

**Review by YuryShkoda — APPROVED** [review]

**Pull request merged**